gio/src/dbus_node_info.rs

```diff
@@ -0,0 +1,38 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use std::ffi::CStr;
+
+use glib::translate::ToGlibPtr;
+
+use crate::{DBusInterfaceInfo, DBusNodeInfo};
+
+impl DBusNodeInfo {
+    pub fn path(self) -> &'static str {
+        let c_obj = self.to_glib_none().0;
+        let path = unsafe { (*c_obj).path };
+        if path.is_null() {
+            return "";
+        }
+        let c_str = unsafe { CStr::from_ptr(path) };
+        c_str.to_str().unwrap()
+    }
+
+    pub fn interfaces(self) -> Vec<DBusInterfaceInfo> {
+        let c_obj = self.to_glib_none().0;
+        let c_ii = unsafe { (*c_obj).interfaces };
+        if c_ii.is_null() {
+            return Vec::new();
+        }
+
+        unsafe { glib::translate::FromGlibPtrContainer::from_glib_none(c_ii) }
+    }
+
+    pub fn nodes(self) -> Vec<DBusNodeInfo> {
```

Comment:
```suggestion
    pub fn nodes<'a>(&'a self) -> &'a [DBusNodeInfo] {
```

as above

Reply:
Done

Reply:
Doing this fails the CI: https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes

Reply:
Oh yes you can omit all the `'a` here, sorry

Reply:
Done